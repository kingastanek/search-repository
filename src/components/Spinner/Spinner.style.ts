import styled, { keyframes } from 'styled-components';

const Load = keyframes`
0%,
80%,
100% {
  box-shadow: 0 0;
  height: 4em;
}
40% {
  box-shadow: 0 -2em;
  height: 5em;
}`;

export const Loader = styled.div`
  color: var(--theme-white);
  text-indent: -9999em;
  margin: 88px auto;
  position: relative;
  font-size: 11px;
  transform: translateZ(0);
  animation-delay: -0.16s;
  background: var(--theme-white);
  animation: ${Load} 1s infinite ease-in-out;
  width: 1em;
  height: 4em;
  &:before,
  &:after {
    background: var(--theme-white);
    animation: ${Load} 1s infinite ease-in-out;
    width: 1em;
    height: 4em;
  }

  &:before,
  &:after {
    position: absolute;
    top: 0;
    content: '';
  }
  &:before {
    left: -1.5em;
    animation-delay: -0.32s;
  }
  &:after {
    left: 1.5em;
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 100;
  background: rgba(0, 0, 0, 0.2);
`;
