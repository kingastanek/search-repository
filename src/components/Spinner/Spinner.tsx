import React from 'react';
import { Wrapper, Loader } from './Spinner.style';

const Spinner = (): JSX.Element => {
  return (
    <Wrapper>
      <Loader />
    </Wrapper>
  );
};

export default Spinner;
