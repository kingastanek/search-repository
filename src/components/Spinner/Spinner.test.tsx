import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './Spinner';
import { Wrapper } from './Spinner.style';

describe('Spinner', () => {
  const baseProps = {
    fixed: true,
  };
  const component = shallow(<Spinner {...baseProps} />);

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should get style for fixed', () => {
    const styled = shallow(<Wrapper fixed />);
    expect(styled).toMatchSnapshot();
  });
  it('should get style for not fixed', () => {
    const styled = shallow(<Wrapper fixed={false} />);
    expect(styled).toMatchSnapshot();
  });
});
