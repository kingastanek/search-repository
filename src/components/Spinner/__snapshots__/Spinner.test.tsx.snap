// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should get style for fixed 1`] = `ShallowWrapper {}`;

exports[`Spinner should get style for not fixed 1`] = `ShallowWrapper {}`;

exports[`Spinner should render correctly 1`] = `ShallowWrapper {}`;
